pkgs/build-support/trivial-builders.nix

```diff
@@ -777,6 +777,36 @@ rec {
         writeDirectReferencesToFile (writeText "string-file" string);
 
 
+  /**
+   * Write the references (i.e. the runtime dependencies in the Nix store) of the elements in `paths' to a file.
+   * The elements are sorted according to their hashes, and duplicated lines are removed.
+   */
+  writeMultipleReferencesToFile = paths: runCommand "runtime-deps-multiple" {
+    # Uniquely sort input paths to reduce unnessesary packages
+    referencesFiles = map writeReferencesToFile (builtins.unique (builtins.sort paths));
```

Comment:
I don't think the unique sort is necessary, `writeReferencesToFile` is deterministic so there won't be any unnecessary paths, and your dictionary below ensure uniqueness.

Reply:
The `lib.unique` here is trying to reduce the build time. `builtins.sort` is probably not needed.

Besides, I think that I might accidentally pushed an older version up here. `builtins.unique` should be `lib.unique`.